modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/CopyFileEntryMVCRenderCommand.java

```diff
@@ -31,6 +47,50 @@
 public class CopyFileEntryMVCRenderCommand
 	extends BaseFileEntryMVCRenderCommand {
 
+	@Override
+	public String render(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws PortletException {
+
+		try {
+			FileEntry fileEntry = ActionUtil.getFileEntry(renderRequest);
+
+			try {
+				_validateCopyToSize(fileEntry);
+
+				return super.render(renderRequest, renderResponse);
+			}
+			catch (FileEntrySizeLimitExceededException
+						fileEntrySizeLimitExceededException) {
+
+				HttpServletRequest originalHttpServletRequest =
+					_portal.getOriginalServletRequest(
+						_portal.getHttpServletRequest(renderRequest));
+
+				SessionErrors.add(
+					originalHttpServletRequest.getSession(),
+					fileEntrySizeLimitExceededException.getClass(),
+					new FileEntrySizeLimitExceededException(
+						DLCopyValidationUtil.getCopyToValidationMessage(
+							_dlSizeLimitConfigurationProvider.
+								getCompanyMaxSizeToCopy(
+									fileEntry.getCompanyId()),
+							_dlSizeLimitConfigurationProvider.
+								getGroupMaxSizeToCopy(fileEntry.getGroupId()),
+							_dlSizeLimitConfigurationProvider.
+								getSystemMaxSizeToCopy(),
+							fileEntry.getSize())));
```

Comment:
This doesn't make sense. You already got an instance of `FileEntrySizeLimitExceededException` in scope, why are you creating a new one? Make sure you initialize the exception correctly when throwing it.